src/Peering/Peering/PeerAsn/RemoveAzurePeerAsnCommand.cs

```diff
@@ -0,0 +1,144 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+namespace Microsoft.Azure.PowerShell.Cmdlets.Peering.PeerAsn
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Linq;
+    using System.Management.Automation;
+
+    using Microsoft.Azure.Commands.Peering.Properties;
+    using Microsoft.Azure.Management.Peering;
+    using Microsoft.Azure.Management.Peering.Models;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Common;
+    using Microsoft.Azure.PowerShell.Cmdlets.Peering.Models;
+    using Microsoft.Rest.Azure;
+
+    using Newtonsoft.Json;
+
+    /// <inheritdoc />
+    /// <summary>
+    ///     New Azure InputObject Command-let
+    /// </summary>
+    [Cmdlet(
+        VerbsCommon.Remove,
+        "AzPeerAsn",
+        SupportsShouldProcess = true,
+        DefaultParameterSetName = Constants.ParameterSetNameDefault)]
+    [OutputType(typeof(bool))]
+    public class RemoveAzurePeerAsn : PeeringBaseCmdlet
+    {
+        /// <summary>
+        ///     Gets or sets The InputObject name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            HelpMessage = Constants.PeerAsnHelp,
+            ParameterSetName = Constants.ParameterSetNameDefault)]
+        [ValidateNotNullOrEmpty]
+        public PSPeerAsn InputObject { get; set; }
+
+        /// <summary>
+        ///     Gets or sets The InputObject name
+        /// </summary>
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            HelpMessage = Constants.PeeringNameHelp,
+            ParameterSetName = Constants.ParameterSetNameByName)]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        /// <summary>
+        ///  Gets or sets the Force the execution of the command.
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = Constants.ForceHelp)]
+        public SwitchParameter Force { get; set; }
```

Comment:
@rileymckenna we should see if we actually need this `-Force` parameter -- typically, we avoid having a `-Force` parameter as it's only used by cmdlets that require additional prompt. For example, if your cmdlet deleted a directory, you would have an additional prompt for the user to confirm if they want to delete everything in that directory as well. The -`Force` parameter in this case will bypass the prompt and go ahead with deleting everything in that folder. More information about this can be found [here](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/design-guidelines/cmdlet-best-practices.md#when-to-add-the-force-parameter).

If it's the case that we need an additional prompt, then having this parameter is fine. If we don't need an additional prompt, then we should look into removing it.

Reply:
our Remove-AzPeerAsn has an internal check that throws a 400 error if they have objects referencing the PeerAsn object. If they dont have any references then it will delete it immediately. I think the ConfirmAction is necessary to ensure users know what they are doing especially since we hide the PeerAsn behind feature flag and have an internal manual check to verify the user. 